docs/configuration/configuration.md

```diff
@@ -252,9 +252,15 @@ metric_relabel_configs:
   [ - <relabel_config> ... ]
 
 # Per-scrape limit on number of scraped samples that will be accepted.
-# If more than this number of samples are present after metric relabelling
+# If more than this number of samples are present after metric relabeling
 # the entire scrape will be treated as failed. 0 means no limit.
 [ sample_limit: <int> | default = 0 ]
+
+# Per-scrape config limit on number of unique targets that will be accepted.
+# If more than this number of targets are present after the relabeling phase,
+# Prometheus will mark the targets as failed without scraping them.
+# 0 means no limit.
```

Comment:
We should probably mark this as experimental in case we want to change this to making the targets disappear in future.